pandas/core/internals.py

```diff
@@ -2388,9 +2388,15 @@ def get_values(self, dtype=None):
         # return object dtype as Timestamps with the zones
         if is_object_dtype(dtype):
             f = lambda x: lib.Timestamp(x, tz=self.values.tz)
-            return lib.map_infer(
+            values = lib.map_infer(
                 self.values.ravel(), f).reshape(self.values.shape)
-        return self.values
+
+            if values.ndim == self.ndim - 1:
```

Comment:
you *can* reshape, but that doesn't belong here. we are trying to make the code simpler, not more complex.

The issue is that you are dealing with a categorical (and for that matter a DatetimeIndex), which by-definition are *always* 1-d objects.

The blocks need to emulate a 2-d structure for compat.

If you find that you need to reshape, need to find another way.